alerting/src/main/kotlin/org/opensearch/alerting/settings/AlertingSettings.kt

```diff
@@ -156,5 +183,155 @@ class AlertingSettings {
+        )
+        val TOTAL_MAX_ACTIONS_ACROSS_TRIGGERS = Setting.intSetting(
+            "plugins.alerting.total_max_actions_across_triggers",
+            DEFAULT_TOTAL_MAX_ACTIONS_ACROSS_TRIGGERS,
+            -1, TotalMaxActionsAcrossTriggersValidator(internalClient),
+            Setting.Property.NodeScope, Setting.Property.Dynamic
+        )
+
+        internal class TotalMaxActionsAcrossTriggersValidator(val client: Client?) : Setting.Validator<Int> {
+            override fun validate(value: Int) {}
+
+            override fun validate(value: Int, settings: Map<Setting<*>, Any>) {
+                val maxActions = settings[TOTAL_MAX_ACTIONS_PER_TRIGGER] as Int
+                validateActionsAcrossTriggers(maxActions, value, client)
+            }
+
+            override fun settings(): MutableIterator<Setting<*>> {
+                val settings = mutableListOf<Setting<*>>(
+                    TOTAL_MAX_ACTIONS_PER_TRIGGER
+                )
+                return settings.iterator()
+            }
+        }
+
+        internal class MaxActionsPerTriggersValidator(val client: Client?) : Setting.Validator<Int> {
+            override fun validate(value: Int) {}
+
+            override fun validate(value: Int, settings: Map<Setting<*>, Any>) {
+                val totalMaxActions = settings[TOTAL_MAX_ACTIONS_ACROSS_TRIGGERS] as Int
+                validateActionsPerTrigger(value, totalMaxActions, client)
+            }
+
+            override fun settings(): MutableIterator<Setting<*>> {
+                val settings = mutableListOf<Setting<*>>(
+                    TOTAL_MAX_ACTIONS_ACROSS_TRIGGERS
+                )
+                return settings.iterator()
+            }
+        }
+
+        private fun validateActionsAcrossTriggers(maxActions: Int, totalMaxActions: Int, client: Client?) {
+            if (totalMaxActions == DEFAULT_TOTAL_MAX_ACTIONS_ACROSS_TRIGGERS) return
+
+            if (maxActions > totalMaxActions) {
+                throw IllegalArgumentException(
+                    "The limit number of actions for a single trigger, $maxActions, " +
+                        "should not be greater than that of the overall max actions across all triggers of the monitor, $totalMaxActions"
+                )
+            }
+            client?.let {
+                GlobalScope.launch {
+                    val monitors = getMonitors(client)
+                    val triggers = getTriggers(monitors)
+
+                    var currentAmountOfActions = getCurrentAmountOfActions(triggers)
+                    currentAmountOfActions += maxActions
+
+                    if (currentAmountOfActions > totalMaxActions)
+                        throw IllegalArgumentException(
+                            "The amount of actions that the client wants to update plus the amount of actions that " +
+                                "already exist, $currentAmountOfActions should not be greater than  that of the " +
```

Comment:
From a security perspective, should we be including `current amount of actions` and `total max actions` in an error message? This could expose information to a user who might not otherwise have privileges to view that info.